dipy/reconst/dki_micro.py

```diff
@@ -0,0 +1,507 @@
+        """ Initialize a KurtosisMicrostrutureModel class instance [1]_.
+
+        Parameters
+        ----------
+        gtab : GradientTable class instance
+
+        fit_method : str or callable
+            str can be one of the following:
+            'OLS' or 'ULLS' to fit the diffusion tensor and kurtosis tensor
+            using the ordinary linear least squares solution
+                dki.ols_fit_dki
+            'WLS' or 'UWLLS' to fit the diffusion tensor and kurtosis tensor
+            using the ordinary linear least squares solution
+                dki.wls_fit_dki
+
+            callable has to have the signature:
+                fit_method(design_matrix, data, *args, **kwargs)
+
+        args, kwargs : arguments and key-word arguments passed to the
+           fit_method. See dki.ols_fit_dki, dki.wls_fit_dki for details
+
+        References
+        ----------
+        .. [1] Fieremans, E., Jensen, J.H., Helpern, J.A., 2011. White Matter
+               Characterization with Diffusion Kurtosis Imaging. Neuroimage
+               58(1): 177-188. doi:10.1016/j.neuroimage.2011.06.006
+        """
+        DiffusionKurtosisModel.__init__(self, gtab, fit_method="WLS", *args,
+                                        **kwargs)
+
+    def fit(self, data, mask=None, sphere='repulsion100', gtol=1e-2,
+            awf_only=False):
+        """ Fit method of the Diffusion Kurtosis Microstructural Model
+
+        Parameters
+        ----------
+        data : array
+            An 4D matrix containing the diffusion-weighted data.
+
+        mask : array
+            A boolean array used to mark the coordinates in the data that
+            should be analyzed that has the shape data.shape[-1]
+
+        sphere : Sphere class instance, optional
+            The sphere providing sample directions for the initial search of
+            the maximal value of kurtosis.
+
+        gtol : float, optional
+            This input is to refine kurtosis maxima under the precision of the
+            directions sampled on the sphere class instance. The gradient of
+            the convergence procedure must be less than gtol before successful
+            termination. If gtol is None, fiber direction is directly taken
+            from the initial sampled directions of the given sphere object
+
+        awf_only : bool, optiomal
+            If set to true only the axonal volume fraction is computed from
+            the kurtosis tensor. Default = False
+        """
+        if mask is None:
+            # Flatten it to 2D either way:
+            data_in_mask = np.reshape(data, (-1, data.shape[-1]))
+        else:
+            # Check for valid shape of the mask
+            if mask.shape != data.shape[:-1]:
+                raise ValueError("Mask is not the same shape as data.")
+            mask = np.array(mask, dtype=bool, copy=False)
+            data_in_mask = np.reshape(data[mask], (-1, data.shape[-1]))
```

Comment:
It just occurred to me that `arr[None]` indexes into the entire array, so we can get rid of this whole `if...else` construct. I will make a PR with that (and another couple of small things here).